src/Hosting/WindowsService/test/UseServiceBaseLifetimeTests.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting.Internal;
+using Microsoft.Extensions.Hosting.WindowsService;
+using Xunit;
+
+namespace Microsoft.Extensions.Hosting
+{
+    public class UseServiceBaseLifetimeTests
+    {
+        [Fact]
+        public void CanOptOutViaConfig()
+        {
+            var host = new HostBuilder()
+                .ConfigureAppConfiguration(config =>
+                {
+                    config.AddInMemoryCollection(new[]
+                    {
+                        new KeyValuePair<string, string>("service", "false")
+                    });
+                })
+                .UseServiceBaseLifetime()
+                .Build();
+
+            using (host)
+            {
+                var lifetime = host.Services.GetRequiredService<IHostLifetime>();
+                Assert.IsType<ConsoleLifetime>(lifetime);
+            }
+        }
+
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX, "Only available on Windows")]
```

Comment:
Remove the string. It's being interpreted as a version string which causes the condition to fail.

Reply:
*head->desk* Thank you. This is why we always use named parameters.